src/HotChocolate/ElasticSearch/src/Data.ElasticSearch/ElasticSearchOperationRewriter.cs

```diff
@@ -61,15 +61,76 @@ protected override IQuery Rewrite(MatchOperation operation)
     }
 
     /// <inheritdoc />
-    protected override IQuery Rewrite(RangeOperation operation)
+    protected override IQuery Rewrite(RangeOperation<double> operation)
     {
-        throw new System.NotImplementedException();
+        return new NumericRangeQuery
+        {
+            Field = operation.Path,
+            GreaterThan = operation.GreaterThan?.Value,
+            LessThan = operation.LowerThan?.Value,
+            GreaterThanOrEqualTo = operation.GreaterThanOrEquals?.Value,
+            LessThanOrEqualTo = operation.LowerThanOrEquals?.Value
+        };
+    }
+
+    /// <inheritdoc />
+    protected override IQuery Rewrite(RangeOperation<long> operation)
+    {
+        return new LongRangeQuery
+        {
+            Field = operation.Path,
+            GreaterThan = operation.GreaterThan?.Value,
+            LessThan = operation.LowerThan?.Value,
+            GreaterThanOrEqualTo = operation.GreaterThanOrEquals?.Value,
+            LessThanOrEqualTo = operation.LowerThanOrEquals?.Value
+        };
+    }
+
+    /// <inheritdoc />
+    protected override IQuery Rewrite(RangeOperation<DateTime> operation)
+    {
+        return new DateRangeQuery
+        {
+            Field = operation.Path,
+            GreaterThan = operation.GreaterThan?.Value,
+            LessThan = operation.LowerThan?.Value,
+            GreaterThanOrEqualTo = operation.GreaterThanOrEquals?.Value,
+            LessThanOrEqualTo = operation.LowerThanOrEquals?.Value
+        };
     }
 
     /// <inheritdoc />
     protected override IQuery Rewrite(TermOperation operation)
     {
-        return new PrefixQuery {Field = operation.Path, Value = operation.Value};
+        return new TermQuery
+        {
+            Field = operation.Path,
+            Value = operation.Value
+        };
+    }
+
+    /// <inheritdoc />
+    protected override IQuery Rewrite(WildCardOperation operation)
+    {
+        return operation.WildCardOperationKind switch
+        {
+            WildCardOperationKind.StartsWith => new WildcardQuery
+            {
+                Field = operation.Path,
+                Wildcard = $"{operation.Value}*"
```

Comment:
I can escape the reserved characters of elastic search. But maybe it would be useful to allow an operation where these are not escaped to enable the api user to use full search capability.

Currently the _WildCard_ operation supports _StartsWith_, _EndsWith_ and _Contains_.

Does it make sense to add e.g. _SearchString_ which is not escaped ?

Reply:
Yes we could have a "search" operation. The question is if this one should be enabled or disabled by default

Reply:
I would say we could enable it because it is just another filter option. It might not work as the user expects but that heavily depends on the user and the index settings.

Reply:
If I should add it I would need an new operation. Currently all operations are set in DefaultFilterOperations. Shall I use a specific one from there, enhance it or should that be done in another ticket?